cpp/src/arrow/util/converter.h

```diff
@@ -294,6 +302,34 @@ class Chunker {
     return status;
   }
 
+  // we could get bit smarter here since the whole batch of appendable values
+  // will be rejected if a capacity error is raised
+  Status Extend(InputType values, int64_t size) {
```

Comment:
The chunker's extend wrapper is untested since we use `Append` from the python code. Theoretically we can wrap the converter's `Extend` method just like in the case of `Append` though we reject the whole batch rather than a single item.

We could improve this logic but would require details about the iteration.

Reply:
I think adding a `int64_t start` in addition to `size` as in #8650 would help. but I haven't yet reviewed how the chunker code works, e.g. can it "know" that it can handle `n` extra values in the current chunk ?